static/js/reload.js

```diff
@@ -129,7 +152,10 @@ export function initialize() {
     }
     ls.remove(hash_fragment);
 
-    [, fragment] = /^#reload:(.*)/.exec(fragment);
+    // TODO: In Sep 2022, `fragment` was changed from a string to a map
+    // containing a string. Once this has been shipped for a bit, we can
+    // remove the `|| fragment`.
```

Comment:
You can use a `TODO/compatibility` comment for this; grep for that and you'll see some examples.

I think the correct statement for when we can safely remove this is something like "Once you can no longer directly upgrade from Zulip 5.x to the current version"; i.e. we're likely to leave this for a year or two. (For Zulip Cloud, it'd be a lot faster, but self-hosters can upgrade in less predictable patterns)

Reply:
Oh nice, thanks!